src/components/auth/registration-form.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import * as z from "zod";
+
+import { SubmitHandler, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import React from "react";
+import Link from 'next/link'
+import { signIn } from 'next-auth/react'
+
+const formSchema = z.object({
+  username: z.string().nonempty("E-mail je povinný").email(),
+  password: z.string().nonempty("Heslo je povinné").min(6, "Heslo musí být dlouhé alespoň 6 znaků"),
+  repeatPassword: z.string().nonempty("Heslo je povinné").min(6, "Heslo musí být dlouhé alespoň 6 znaků"),
+});
+
+
+export type FormValues = z.infer<typeof formSchema>;
+
+export const RegistrationForm = () => {
+  const form = useForm<FormValues>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      username: "",
+      password: "",
+      repeatPassword: "",
+    },
+  });
+
+  const handleSubmit: SubmitHandler<FormValues> = async ({ username, password }, event) => {
+    event?.preventDefault();
+    console.log({ username, password })
+  };
+
+  const errors = form.formState.errors;
+
+  return (
+    <form onSubmit={form.handleSubmit(handleSubmit)} className="space-y-8">
+      <div className="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8">
+        <div className="sm:mx-auto sm:w-full sm:max-w-sm">
+          <img
+            className="mx-auto h-10 w-auto"
+            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
+            alt="Your Company"
+          />
+          <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">
+            Registrovat se
+          </h2>
+        </div>
+
+        <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
+          <div>
+            <label htmlFor="email" className="block text-sm font-medium leading-6 text-gray-900">
+              E-mail
+            </label>
+            <div className="mt-2">
+              <input
+                id="email"
+                type="email"
+                {...form.register("username")}
+                autoComplete="email"
+                required
+                className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
+              />
+              {errors.username?.message && <p className="text-red-400">{errors.username.message}</p>}
```

Comment:
Toto bych udělal jako komponentu. Zde se opakuje několikrát, myslím, že se bude opakovat ještě na mnoha dalších stránkách.

Reply:
Souhlas. Bude to chtít překlopit na komponenty. Nicméně pojďme to dělat iterativně. Toto zamergujme a pak to posléze můžeme překopat na komponenty.